plugins/modules/site_workflow_manager.py

```diff
@@ -760,14 +992,266 @@ def get_want(self, config):
+
+        Parameters:
+            keymap (dict): Already existing key map dictionary to add to or empty dict {}.
+            data (dict): Input data where keys need to be mapped using the key map.
+
+        Returns:
+            dict: A dictionary with the original keys as values and the converted snake_case
+                    keys as keys.
+
+        Example:
+            functions = Accesspoint(module)
+            keymap = functions.map_config_key_to_api_param(keymap, device_data)
+        """
+        if keymap is None:
+            keymap = {}
+
+        if isinstance(data, dict):
+            keymap.update(keymap)
+            for key, value in data.items():
+                new_key = re.sub(r'([a-z0-9])([A-Z])', r'\1_\2', key).lower()
+                keymap[new_key] = key
+
+                if isinstance(value, dict):
+                    self.map_config_key_to_api_param(keymap, value)
+                elif isinstance(value, list):
+                    for item in value:
+                        if isinstance(item, dict):
+                            self.map_config_key_to_api_param(keymap, item)
+        elif isinstance(data, list):
+            for item in data:
+                if isinstance(item, dict):
+                    self.map_config_key_to_api_param(keymap, item)
+
+        return keymap
+
+    def update_floor(self, site_params):
+        """
+        Updates a floor in the site hierarchy using the provided site parameters.
+
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            site_params (dict): Dictionary containing parameters required for the floor update, including the site_id.
+
+        Returns:
+            dict: The API response from the 'updates_a_floor' operation.
+        """
+
+        try:
+            self.log("Updating floor with parameters: {0}".format(site_params), "INFO")
+            parent_name = site_params.get("site", {}).get("floor", {}).get("parentName")
+            type = site_params.get("type")
+            (parent_id) = self.get_parent_id(parent_name)
+            site_params['site']['floor']['parentId'] = parent_id
+            site_params['site']['floor']['unitsOfMeasure'] = "feet"
+            self.log("Updated site_params with parent_id: {0}".format(site_params), "INFO")
+            floor_param = site_params.get('site', {}).get('floor')
+            site_id = self.have.get("site_id")
+            floor_param['id'] = site_id
+            self.log(floor_param)
+
+            response = self.dnac._exec(
+                family="site_design",
+                function='updates_a_floor',
+                op_modifies=True,
+                params=floor_param
+            )
+            self.log(response)
+            if response and isinstance(response, dict):
+                self.log("Initial API response from 'updates_a_floor': {0}".format(response), "DEBUG")
```

Comment:
Don't log same log message multiple times in same API.

Reply:
Removed extra log